Code/IO/include/sitkImageViewer.h

```diff
@@ -0,0 +1,211 @@
+   *
+   *  For example, the default command string on Linux systems is:
+   *
+   *  \code
+   *  %a -o %f
+   *  \endcode
+   *
+   *  After token substitution it may become:
+   *
+   *  \code
+   *  ImageJ -o /tmp/Temp-65535-0.nii
+   *  \endcode
+   *
+   *  For another example, the default command string for color images on Mac OS X is:
+   *
+   *  \code
+   *  open -a %a -n --args -eval \'open(\"%f\"); run(\"Make Composite\", \"display=Composite\"); \'
+   *  \endcode
+   *
+   *  After token substitution the string may become:
+   *
+   *  \code
+   *  open -a ImageJ64 -n --args -eval 'open("/tmp/TempFile-20238-0.nii"); run("Make Composite", "display=Composite");'
+   *  \endcode
+   *
+   *  The string after \c '-eval' is an ImageJ macro the opens the file and runs ImageJ's Make Composite
+   *  command to display the image in color.
+   *
+   *  If the \c '%f' token is not found in the command string, the temporary file name is automatically
+   *  appended to the command argument list.
+   *
+   *  Note: Using the ImageViewer::SetCommand method overrides the default command and/or any environment
+   *  variable setting such as SITK_SHOW_COMMAND, SITK_SHOW_COLOR_COMMAND and SITK_SHOW_3D_COMMAND.
+   * @{
+   */
+  void SetCommand( const std::string & command );
+  const std::string & GetCommand(bool fijiFlag=false, bool colorImage=false, bool threeDImage=false) const;
+  /**@}*/
+
+  /** \brief Set/Get file extension of the temporary image file (default='.nii')
+   *
+   * If the viewing application is Fiji, the default image format is [MetaIO](https://itk.org/Wiki/ITK/MetaIO), '.mha'.
+   *
+   * If the viewing application is ImaegJ, the default image format is [NifTi](https://nifti.nimh.nih.gov/nifti-1/), '.nii'.  This is because by default ImageJ does not come with the MetaIO plugin.
+   * @{
+   */
+  void SetFileExtension( const std::string & ext );
+  const std::string & GetFileExtension() const;
+  /**@}*/
+
+  /** \brief Set/Get Debug flag.
+   *
+   * Turning on the Debug flag produces output showing the command string used to
+   * launch the viewing application.
+   * @{
+   */
+  static void SetDebug( const bool dbg );
+  static bool GetDebug();
+  /**@}*/
+
+  /** \brief Set/Get Process delay, the wait time after launching the viewing application.
+   *
+   * On Windows the delay is in seconds, default=1sec.
+   * On Mac/Linux the delay is in milli-seconds, default=500ms.
+   * @{
+   */
+  static void SetProcessDelay( const unsigned int delay);
+  static unsigned int GetProcessDelay();
+  /**@}*/
+
+  /** \brief Set/Get Title string on the ImageJ window.
+   * @{
+   */
+  void SetTitle( const std::string & t );
+  const std::string & GetTitle() const;
+  /**@}*/
+
+  /** \brief Launch the viewing application with to display the given image.
+   */
+  void Execute ( const Image &image );
+
+  /** \brief Print ourself out to a string.
+   */
+  std::string ToString() const;
+
+
+private:
+
```

Comment:
Let us continue to follow ITK/SimpleITK convention of having member variables names be prefixed with `m_` then camel case.